common/src/main/java/com/wynntils/features/user/overlays/map/MiniMapOverlayFeature.java

```diff
@@ -104,38 +113,100 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
 
             // TODO compass icon
 
-            // disable mask
+            // cursor
+            if (!followPlayerRotation) {
+                poseStack.pushPose();
+                rotateMap(poseStack, centerX, centerZ, 180 + yRot);
+            }
+
+            // TODO cursor color?
+            RenderUtils.drawTexturedRect(
+                    poseStack,
+                    Texture.MAP_POINTERS.resource(),
+                    (int) (centerX - pointerType.width / 2),
+                    (int) (centerZ - pointerType.height / 2),
+                    0,
+                    pointerType.width,
+                    pointerType.height,
+                    0,
+                    pointerType.textureY,
+                    pointerType.width,
+                    pointerType.height,
+                    Texture.MAP_POINTERS.width(),
+                    Texture.MAP_POINTERS.height());
+
+            if (!followPlayerRotation) {
+                poseStack.popPose();
+            }
+
+            // disable mask & render border
             switch (maskType) {
                 case Rectangular -> {
                     RenderSystem.disableScissor();
+                    renderRectangularMapBorder(poseStack, renderX, renderY, width, height);
                 }
                     // case Circle -> {
                     // TODO
+                    // renderCircularMapBorder();
                     // }
             }
 
-            // TODO cursor
-
-            // TODO Border rendering
-
             // TODO Directional Text
 
             // TODO Coords
 
         }
 
-        private void rotateMapToPlayer(PoseStack poseStack, float centerX, float centerZ) {
+        private void rotateMap(PoseStack poseStack, float centerX, float centerZ, float angle) {
```

Comment:
Can we rename this? This does not actually rotate the map, it just does poseStack translations.

Reply:
Perhaps rotatePose?

Reply:
I think I will move the method to `RenderUtils` as well